sentry_sdk/tracing.py

```diff
@@ -14,36 +20,71 @@ def get(self, key):
         return self.environ.get("HTTP_" + key.replace("-", "_").upper())
 
 
-class SpanContext(object):
-    def __init__(self, trace_id, span_id, recorded=False, parent=None):
+class Span(object):
+    __slots__ = (
+        "trace_id",
+        "span_id",
+        "parent_span_id",
+        "same_process_as_parent",
+        "sampled",
+        "transaction",
+        "op",
+        "description",
+        "start_timestamp",
+        "timestamp",
+        "_tags",
+        "_data",
+        "_finished_spans",
+    )
+
+    def __init__(
+        self,
+        trace_id,
+        span_id,
+        parent_span_id=None,
+        same_process_as_parent=True,
+        sampled=None,
+        transaction=None,
+        op=None,
+        description=None,
+    ):
         self.trace_id = trace_id
         self.span_id = span_id
-        self.recorded = recorded
-        self.parent = None
+        self.parent_span_id = parent_span_id
+        self.same_process_as_parent = same_process_as_parent
+        self.sampled = sampled
+        self.transaction = transaction
+        self._tags = {}
+        self._data = {}
+        self._finished_spans = []
+        self.start_timestamp = datetime.now()
+        self.timestamp = None
 
     def __repr__(self):
-        return "%s(trace_id=%r, span_id=%r, recorded=%r)" % (
+        return "<%s(transaction=%r, trace_id=%r, span_id=%r, parent_span_id=%r)>" % (
```

Comment:
Would be nice to output the sampling decision here too

Reply:
Done